### PR TITLE
fix: the recognizer dialog created is for LUIS even if Orchestrator is selected in dropdown

### PR DESCRIPTION
**Pull request by lei9444**
## Description
**In this PR**
 check the dialog recognizer type, if it is Orchestrator recognizer type, generate the recognizer dialog with orchestrator recognizer template

orch

## Task Item
#minor

### CODE REVIEW

File: Composer/packages/client/src/recoilModel/Recognizers.tsx [hunk 1]
```diff
@@ -148,40 +165,43 @@ export const Recognizer = React.memo((props: { projectId: string }) => {
   const qnaFiles = useRecoilValue(qnaFilesState(projectId));
   const settings = useRecoilValue(settingsState(projectId));
   const curRecognizers = useRecoilValue(recognizersSelectorFamily(projectId));
+  const curRecognizersRef = useRef(curRecognizers);
   const filePersistence = useRecoilValue(filePersistenceState(projectId));
+  curRecognizersRef.current = curRecognizers;
```
Comment (taicchoumsft) [inline Composer/packages/client/src/recoilModel/Recognizers.tsx:170]: Not sure if we should move this to the `useEffect` just to be safe?

**Review by taicchoumsft — APPROVED** [review]
Tested and it works as expected.  Signing off.

File: Composer/packages/client/src/recoilModel/Recognizers.tsx [hunk 2]
```diff
@@ -148,40 +165,43 @@ export const Recognizer = React.memo((props: { projectId: string }) => {
...
-        setRecognizers(recognizers);
-      }
+    if (isEmpty(filePersistence)) return;
+    let recognizers: RecognizerFile[] = [];
+
+    dialogs
+      .filter((dialog) => isCrossTrainedRecognizerSet(dialog) || isLuisRecognizer(dialog))
+      .forEach((dialog) => {
+        const luProvide = getLuProvider(dialog.id, curRecognizersRef.current);
+        const filtedLus = luFiles.filter((item) => getBaseName(item.id) === dialog.id);
```
[the remark this answers is not in the document]
Comment (lei9444) [reply Composer/packages/client/src/recoilModel/Recognizers.tsx]: Thanks @taicchoumsft , I think I need to improve my english

File: Composer/packages/client/src/recoilModel/Recognizers.tsx (hunk 1 above)
Comment (lei9444) [reply Composer/packages/client/src/recoilModel/Recognizers.tsx:168]: I'm afraid that the setRecognizer will trigger the  calculation again if I put it to the useEffect dependency

**Review by boydc2014 — APPROVED** [review]

**Pull request merged**